model/opinions.go

```diff
@@ -38,7 +40,61 @@ func newOpinions(lightFile, darkFile string) (*opinions, error) {
 	return result, nil
 }
 
-func (o *opinions) GetOpinionForKey(opinions map[string]interface{}, keyPieces []string) (result interface{}) {
+// FlatMap converts the incoming nested map of opinions into a flat
+// map of properties to values (strings).
+func FlatMap(opinions map[string]interface{}) map[string]string {
```

Comment:
This transform is quite specific to the opinions IMHO, would leave it here.
Yes to tests.
Yes to better name.